### PR TITLE
BUG: unexpected error from `DataFrame.__mars_tensor__`

### LINKED ISSUE #339: BUG: `Dataframe.__mars_tensor__` raises exception if df contains a string
### Describe the bug
`Dataframe.__mars_tensor__` should convert the dataframe into a tensor with given dtype. If dtype is not specified, it should be inferred from the dataframe's dtypes.

But currently, if dtype is absent and the dataframe contains a string, an exception will be raised.

### To Reproduce
To help us to reproduce this bug, please provide information below:

1. Your Python version: `3.9.15`
3. The version of Xorbits you use: `0.2.1`
5. Full stack of the error:
```python
```python
ValueError                                Traceback (most recent call last)
File ~/miniconda3/envs/gpu/lib/python3.9/site-packages/IPython/core/formatters.py:706, in PlainTextFormatter.__call__(self, obj)
    699 stream = StringIO()
    700 printer = pretty.RepresentationPrinter(stream, self.verbose,
    701     self.max_width, self.newline,
    702     max_seq_length=self.max_seq_length,
    703     singleton_pprinters=self.singleton_printers,
    704     type_pprinters=self.type_printers,
    705     deferred_pprinters=self.deferred_printers)
--> 706 printer.pretty(obj)
    707 printer.flush()
    708 return stream.getvalue()

File ~/miniconda3/envs/gpu/lib/python3.9/site-packages/IPython/lib/pretty.py:410, in RepresentationPrinter.pretty(self, obj)
    407                         return meth(obj, self, cycle)
    408                 if cls is not object \
    409                         and callable(cls.__dict__.get('__repr__')):
--> 410                     return _repr_pprint(obj, self, cycle)
    412     return _default_pprint(obj, self, cycle)
    413 finally:

File ~/miniconda3/envs/gpu/lib/python3.9/site-packages/IPython/lib/pretty.py:778, in _repr_pprint(obj, p, cycle)
    776 """A pprint that just redirects to the normal repr function."""
    777 # Find newlines and replace them with p.break_()
--> 778 output = repr(obj)
    779 lines = output.splitlines()
    780 with p.group():

File ~/repos/xorbits/python/xorbits/utils.py:37, in safe_repr_str.<locals>.inn(self, *args, **kwargs)
     35     return getattr(object, f.__name__)(self)
     36 else:
---> 37     return f(self, *args, **kwargs)

File ~/repos/xorbits/python/xorbits/core/data.py:253, in DataRef.__repr__(self)
    251     return self.data._mars_entity.op.data.__repr__()
    252 else:
--> 253     run(self)
    254     return self.data.__repr__()

File ~/repos/xorbits/python/xorbits/core/execution.py:55, in run(obj, **kwargs)
     53 mars_tileables = [_get_mars_entity(ref) for ref in refs_to_execute.values()]
     54 if mars_tileables:
---> 55     mars_execute(mars_tileables, **kwargs)

File ~/repos/xorbits/python/xorbits/_mars/deploy/oscar/session.py:1877, in execute(tileable, session, wait, new_session_kwargs, show_progress, progress_update_interval, *tileables, **kwargs)
   1875     session = get_default_or_create(**(new_session_kwargs or dict()))
   1876 session = _ensure_sync(session)
-> 1877 return session.execute(
   1878     tileable,
   1879     *tileables,
   1880     wait=wait,
   1881     show_progress=show_progress,
   1882     progress_update_interval=progress_update_interval,
   1883     **kwargs,
   1884 )

File ~/repos/xorbits/python/xorbits/_mars/deploy/oscar/session.py:1671, in SyncSession.execute(self, tileable, show_progress, warn_duplicated_execution, *tileables, **kwargs)
   1669 fut = asyncio.run_coroutine_threadsafe(coro, self._loop)
   1670 try:
-> 1671     execution_info: ExecutionInfo = fut.result(
   1672         timeout=self._isolated_session.timeout
   1673     )
   1674 except KeyboardInterrupt:  # pragma: no cover
   1675     logger.warning("Cancelling running task")

File ~/miniconda3/envs/gpu/lib/python3.9/concurrent/futures/_base.py:446, in Future.result(self, timeout)
    444     raise CancelledError()
    445 elif self._state == FINISHED:
--> 446     return self.__get_result()
    447 else:
    448     raise TimeoutError()

File ~/miniconda3/envs/gpu/lib/python3.9/concurrent/futures/_base.py:391, in Future.__get_result(self)
    389 if self._exception:
    390     try:
--> 391         raise self._exception
    392     finally:
    393         # Break a reference cycle with the exception in self._exception
    394         self = None

File ~/repos/xorbits/python/xorbits/_mars/deploy/oscar/session.py:1857, in _execute(session, wait, show_progress, progress_update_interval, cancelled, *tileables, **kwargs)
   1854     else:
   1855         # set cancelled to avoid wait task leak
   1856         cancelled.set()
-> 1857     await execution_info
   1858 else:
   1859     return execution_info

File ~/miniconda3/envs/gpu/lib/python3.9/asyncio/tasks.py:688, in _wrap_awaitable(awaitable)
    681 @types.coroutine
    682 def _wrap_awaitable(awaitable):
    683     """Helper for asyncio.ensure_future().
    684
    685     Wraps awaitable (an object with __await__) into a coroutine
    686     that will later be wrapped in a Task by ensure_future().
    687     """
--> 688     return (yield from awaitable.__await__())

File ~/repos/xorbits/python/xorbits/_mars/deploy/oscar/session.py:104, in ExecutionInfo._ensure_future.<locals>.wait()
    103 async def wait():
--> 104     return await self._aio_task

File ~/repos/xorbits/python/xorbits/_mars/deploy/oscar/session.py:952, in _IsolatedSession._run_in_background(self, tileables, task_id, progress, profiling)
    946         logger.warning(
    947             "Profile task %s execution result:\n%s",
    948             task_id,
    949             json.dumps(task_result.profiling, indent=4),
    950         )
    951     if task_result.error:
--> 952         raise task_result.error.with_traceback(task_result.traceback)
    953 if cancelled:
    954     return

File ~/repos/xorbits/python/xorbits/_mars/services/task/supervisor/processor.py:388, in TaskProcessor.run(self)
    386 async with self._executor:
    387     async for stage_args in self._iter_stage_chunk_graph():
--> 388         await self._process_stage_chunk_graph(*stage_args)
    389 await self._task_info_collector.collect_result_nodes(
    390     self._task, self._subtask_graphs
    391 )
    392 await self._task_info_collector.collect_tileable_structure(
    393     self._task, self.get_tileable_to_subtasks()
    394 )

File ~/repos/xorbits/python/xorbits/_mars/services/task/supervisor/processor.py:265, in TaskProcessor._process_stage_chunk_graph(self, stage_id, stage_profiler, chunk_graph)
    259 tile_context = await asyncio.to_thread(
    260     self._get_stage_tile_context,
    261     {c for c in chunk_graph.result_chunks if not isinstance(c.op, Fetch)},
    262 )
    264 with Timer() as timer:
--> 265     chunk_to_result = await self._executor.execute_subtask_graph(
    266         stage_id, subtask_graph, chunk_graph, tile_context
    267     )
    268 stage_profiler.set("run", timer.duration)
    270 self._preprocessor.post_chunk_graph_execution()

File ~/repos/xorbits/python/xorbits/_mars/services/task/execution/mars/executor.py:203, in MarsTaskExecutor.execute_subtask_graph(self, stage_id, subtask_graph, chunk_graph, tile_context, context)
    201 curr_tile_progress = self._tile_context.get_all_progress() - prev_progress
    202 self._stage_tile_progresses.append(curr_tile_progress)
--> 203 return await stage_processor.run()

File ~/repos/xorbits/python/xorbits/_mars/services/task/execution/mars/stage.py:233, in TaskStageProcessor.run(self)
    229     if self.subtask_graph.num_shuffles() > 0:
    230         # disable scale-in when shuffle is executing so that we can skip
    231         # store shuffle meta in supervisor.
    232         await self._scheduling_api.disable_autoscale_in()
--> 233     return await self._run()
    234 finally:
    235     if self.subtask_graph.num_shuffles() > 0:

File ~/repos/xorbits/python/xorbits/_mars/services/task/execution/mars/stage.py:253, in TaskStageProcessor._run(self)
    251 if self.error_or_cancelled():
    252     if self.result.error is not None:
--> 253         raise self.result.error.with_traceback(self.result.traceback)
    254     else:
    255         raise asyncio.CancelledError()

File ~/repos/xorbits/python/xorbits/_mars/services/subtask/worker/processor.py:210, in _execute_operand()
    205 @enter_mode(build=False, kernel=True)
    206 def _execute_operand(
    207     self, ctx: Dict[str, Any], op: OperandType
    208 ):  # noqa: R0201  # pylint: disable=no-self-use
    209     try:
--> 210         return execute(ctx, op)
    211     except BaseException as ex:
    212         # wrap exception in execution to avoid side effects
    213         raise ExecutionError(ex).with_traceback(ex.__traceback__) from None

File ~/repos/xorbits/python/xorbits/_mars/core/operand/core.py:492, in execute()
    488 else:
    489     # Cast `UFuncTypeError` to `TypeError` since subclasses of the former is unpickleable.
    490     # The `UFuncTypeError` was introduced by numpy#12593 since v1.17.0.
    491     try:
--> 492         result = executor(results, op)
    493         succeeded = True
    494         if op.stage is not None:

File ~/repos/xorbits/python/xorbits/_mars/tensor/base/astype.py:86, in execute()
     84 else:
     85     if is_same_module(xp, np):
---> 86         ctx[chunk.key] = x.astype(
     87             op.dtype, order=op.order, casting=op.casting
     88         )
     89     else:  # pragma: no cover
     90         # cupy does not support casting
     91         ctx[chunk.key] = x.astype(op.dtype, order=op.order)

ValueError: could not convert string to float: 'foo'
```
6. Minimized code to reproduce the error.
```python
df = pd.DataFrame(["foo", "bar", "baz"])
df.__mars_tensor__()
```

### Expected behavior
```python
array([['foo'],
       ['bar'],
       ['baz']], dtype=object)
```

### PR DESCRIPTION
**Pull request by UranusSeven**
## What do these changes do?

Fixes #339 

## Check code requirements

- [x] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

File: python/xorbits/_mars/dataframe/core.py [hunk 1]
```diff
@@ -2183,7 +2183,9 @@ def __repr__(self):
         return self._to_str(representation=True)
 
     def __mars_tensor__(self, dtype=None, order="K"):
-        return self.to_tensor().astype(dtype=dtype, order=order, copy=False)
+        tensor = self.to_tensor()
+        dtype = dtype if dtype is not None else tensor.dtype
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/core.py:2187]: If `dtype` is None, do we need to call `astype`?

Comment (qianduoduo0904) [reply python/xorbits/_mars/dataframe/core.py:2187]: The order could be different.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**